lib/CustomFields/utils/useSectionTitleFetch.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect, useCallback } from 'react';
+
+import makeRequest from './makeRequest';
+
+const useSectionTitleFetch = (okapi, moduleName) => {
+  const [sectionTitleLoading, setSectionTitleLoading] = useState(false);
+  const [sectionTitle, setSectionTitle] = useState({});
+  const [sectionTitleHasError, setSectionTitleHasError] = useState(false);
+
+  const makeOkapiRequest = useCallback(
+    makeRequest(okapi)(),
+    [okapi.token, okapi.tenant, okapi.url]
+  );
+
+  useEffect(() => {
+    const fetchCustomFields = async () => {
+      setSectionTitleLoading(true);
+
+      const response = await makeOkapiRequest(`configurations/entries?query=(module==${moduleName} and configName==custom_fields_label)`)({
+        method: 'GET',
+      });
+
+      if (response.ok) {
+        const { configs } = await response.json();
+
+        setSectionTitle({ ...configs[0] });
+      } else {
+        setSectionTitleHasError(true);
+      }
+    };
+
+    if (!sectionTitleLoading) {
+      fetchCustomFields();
+    }
+  });
```

Comment:
please, add all of the necessary dependencies to the effect. now it runs after every component update